libsupport/include/katana/OpaqueID.h

```diff
@@ -241,6 +241,11 @@ struct OpaqueIDLinear : public OpaqueIDOrderedWithValue<_IDType, _Value> {
     return *static_cast<_IDType*>(this);
   }
 
+  _IDType& operator+=(const _IDType& rhs) {
+    this->value_ += rhs.value();
+    return *static_cast<_IDType*>(this);
+  }
```

Comment:
This isn't correct theoretically. The issue is that Linear Opaque IDs are *not* numbers. They are ordered and have differences, but they have no "value" inherently (`value()` is a backdoor). If you want to do a prefix sum, you should do it on the `DifferenceType` not on the IDs themselves. Pick a starting ID and compute differences out from that. This makes it explicit when you are working on numerical differences instead of IDs themselves (which again are not numbers and have no numeric value).

All that said, we can add this, but I don't think it should be on `Linear` since this changes that semantics of the OpaqueID it's used on to combine the idea of IDs and differences. I think that should be a separate subclass. Maybe we could call it `NumericID`. It would not have a difference type and would return IDs from subtraction as well as supporting addition of IDs.

Reply:
I agree with this. After spending some time with `OpaqueID`s I have been convinced that differentiating between `DifferenceType` and `ValueType` is a valuable feature.

Reply:
> This isn't correct theoretically. The issue is that Linear Opaque IDs are _not_ numbers. They are ordered and have differences, but they have no "value" inherently (`value()` is a backdoor). If you want to do a prefix sum, you should do it on the `DifferenceType` not on the IDs themselves. Pick a starting ID and compute differences out from that. This makes it explicit when you are working on numerical differences instead of IDs themselves (which again are not numbers and have no numeric value).
 
Makes sense; I'll add another type. I'm using these as typed indexes so I'm implicitly ascribing numeric qualities to them already.

Reply:
That makes sense. Maybe `IndexID` would be a good name for it. Since that use case is common and makes sense.